src/common/browser-adapters/browser-adapter-event-manager.ts

```diff
@@ -76,19 +95,27 @@ export class BrowserAdapterEventManager {
             result = listener(...eventArgs);
         } catch (error) {
             console.error('Error thrown in application listener: ', error);
-            throw error;
+            result = error;
         }
         if (!!result && typeof result.then === 'function') {
-            //wrapping application listener promise responses in a 4-minute promise race
+            //wrapping application listener promise responses in a 4-minute promise resolution
             //prevents the service worker going idle before a response is sent
-            return this.promiseFactory.timeout(result, FOUR_MINUTES);
+            return this.promiseFactory.delay(result, FOUR_MINUTES);
```

Comment:
I think `delay` is right for the behavior on L108, but I don't think it's quite what what we want here - if an application handler returns a `Promise` that resovles in a timely fashion, I think we should probably trust it to encapsulate the required async work and resolve as soon as it does, not delay it for 4 minutes. I think the 4 minutes should be a limit, not a minimum delay.

I think a straightforward way to achieve that but still avoid rejecting the promise would be something like

```ts
return this.promiseFactory.timeout(result, FOUR_MINUTES)
    .catch(() => {
        // We want to avoid rejecting the Promise we give back to the browser event handler
        // because doing so is liable to cause the browser to tear down our Service Worker
        // immediately, even if other events are in progress.
        return undefined;
    });
```